Pantsaclysm, Trouser Variants Ahead.

DoctorBoomstick opened the pull request:
#### Summary
Content "adds cosmetic variants for various items, including regular pants, fur coats, high-heels, and knee-length boots, as well as also adding tactical pants to the game."
#### Purpose of change
Pants make the world go around. CDDA’s world is currently going around in a never-ending death spiral. One plus one equals we needed pants variants. This PR seeks to add such pants variants because variants are fun, generic items are not, and we need to build up the variant numbers further.
#### Describe the solution
These additions were originally going to be bundled in another PR I’m working on; however, I wanted to get the variants in game, and I figured that reviewing a 1500-plus line PR isn’t fun for anybody, so I broke them off into their own pull request.
This PR adds:
- Several boat-loads of pants variants.
- Several variants for the generic fur coat.
- A few variants each for the high-heels and knee-length boots.
- A new tactical pants item with it’s own cosmetic variants.

Several professions have been tweaked in order to ensure that they spawn with sensible styles of pants (no hot pink sweatpants for lab technicians), while the sable fur coat has been deprecated and removed.
#### Describe alternatives you've considered
Leaving these variants back in my original PR, adding more variants to the complement already present, and working on something tangibly important to the game’s development—not that there’s anything more vital than pants, of course—
#### Testing

#### Additional context
Thanks to Cosmic on the Discord for providing me with the list of pants colors, for which I simply filled in the descriptions.

https://discord.com/channels/598523535169945603/598523535169945607/1114945458415075348

DoctorBoomstick committed:
Adds the stuff.

DoctorBoomstick committed:
bug fix.

TheShadowFerret commented on data/json/items/armor/boots.json:
```diff
@@ -1503,7 +1537,28 @@
+      {
+        "id": "chelsea_boots",
+        "name": { "str": "pair of chelsea boots", "str_pl": "pairs of chelsea boots" },
+        "description": "A pair of knee-length, laceless boots formed from calfskin with an elastic side panel that’s designed to stretch away to facilitate simple dawning and doffing, these stylish stompers are a time-old gift from Victorian-age England.  Intended for you to fashionably saunter down roadways rather than sprint for your life, this style of boot became a style staple within London  ...
```
```suggestion
        "description": "A pair of knee-length, laceless boots formed from calfskin with an elastic side panel that’s designed to stretch away to facilitate simple dawning and doffing, these stylish stompers are a time-old gift from Victorian-age England.  Intended for you to fashionably saunter down roadways rather than sprint for your life, this style of boot became a style staple within London during the 1960s, where they were synonymous with the mod crowd.",
```

TheShadowFerret commented on data/json/items/armor/coats.json:
```diff
@@ -91,28 +91,123 @@
+      {
+        "id": "coat_fox_arctic_fur",
+        "name": { "str": "arctic fox jacket" },
+        "description": "Fluffy and insulating, this expensive fur coat is composed of a small battalion’s worth of snowy-white arctic fox pelts with silver tips.  A strait section forms the back and lower sweep, while triangularly cut furs are sown in an inverse pattern to form the rest of the jackit.",
```
```suggestion
        "description": "Fluffy and insulating, this expensive fur coat is composed of a small battalion’s worth of snowy-white arctic fox pelts with silver tips.  A straight section forms the back and lower sweep, while triangularly cut furs are sown in an inverse pattern to form the rest of the jacket.",
```

TheShadowFerret commented on data/json/items/armor/coats.json:
```diff
@@ -91,28 +91,123 @@
+      {
+        "id": "coat_fox_frost_fur",
+        "name": { "str": "frost fox overcoat" },
+        "description": "A strait-cut, classically stylish jacket crafted from soft and voluminous gray fox fur with a blue undercoat, this article of winter apparel comes kitted out with a tall collar and two modest pockets.",
```
```suggestion
        "description": "A straight-cut, classically stylish jacket crafted from soft and voluminous gray fox fur with a blue undercoat, this article of winter apparel comes kitted out with a tall collar and two modest pockets.",
```

TheShadowFerret commented on data/json/items/armor/coats.json:
```diff
@@ -91,28 +91,123 @@
+      {
+        "id": "bomber_fox_fur",
+        "name": { "str": "fox fur bomber jacket" },
+        "description": "What was once a rugged piece of military clothing has metamorphosed into a fashion icon within this slim-cut and insulated jacket, with wide, cinchable sleeves and a broad collar.  Composed of sown-together fox pelts and fastened with a zipper at the front, the inside of this jacket is lined with plush, reddish orange fur.",
```
```suggestion
        "description": "What was once a rugged piece of military clothing has metamorphosed into a fashion icon within this slim-cut and insulated jacket, with wide, cinchable sleeves and a broad collar.  Composed of sewn-together fox pelts and fastened with a zipper at the front, the inside of this jacket is lined with plush, reddish orange fur.",
```

TheShadowFerret commented on data/json/items/armor/coats.json:
```diff
@@ -91,28 +91,123 @@
+      {
+        "id": "coat_chinchilla _fur",
+        "name": { "str": "chinchilla jacket" },
+        "description": "Luxuriously velvety and ridiculously extra, this warm, gray-hued coat is ribbed with patterns of black and white highlights.  Composed of a small army’s worth of dead Chinchilla pelts and naturally hypoallergenic, the world might be going to hell, but at least you’ll look like a celebrity while enveloped within this dreamily soft, timelessly stylish coat.",
```
```suggestion
        "description": "Luxuriously velvety and ridiculously extra, this warm, gray-hued coat is ribbed with patterns of black and white highlights.  Composed of a small army’s worth of dead chinchilla pelts and naturally hypoallergenic, the world might be going to hell, but at least you’ll look like a celebrity while enveloped within this dreamily soft, timelessly stylish coat.",
```

TheShadowFerret commented on data/json/items/armor/coats.json:
```diff
@@ -91,28 +91,123 @@
+      {
+        "id": "coat_rabbit_fur",
+        "name": { "str": "rabbit overcoat" },
+        "description": "Prey and predator are brought together within this luxurious fir coat, sown from panels of velvety, snowy-white rabbit pelts and accentuated by a beige layer of fox fur trimming the collar and cuffs.  With a tapered waist and two side pockets, the jacket is both comfortable to wear and stylish.",
```
```suggestion
        "description": "Prey and predator are brought together within this luxurious fir coat, sewn from panels of velvety, snowy-white rabbit pelts and accentuated by a beige layer of fox fur trimming the collar and cuffs.  With a tapered waist and two side pockets, the jacket is both comfortable to wear and stylish.",
```

TheShadowFerret commented on data/json/items/armor/coats.json:
```diff
@@ -91,28 +91,123 @@
+      {
+        "id": "coat_albino_mink_fur",
+        "name": { "str": "albino mink coat" },
+        "description": "Even more prized than regular mink jackets, this coat is exclusively composed of plush white fur with no markings or color in its undercoat.  Extremely warm and equipped with dual pockets, this expensive fashion statement would certainly turn some heads if anyone left alive could still care about esthetics.",
```
```suggestion
        "description": "Even more prized than regular mink jackets, this coat is exclusively composed of plush white fur with no markings or color in its undercoat.  Extremely warm and equipped with dual pockets, this expensive fashion statement would certainly turn some heads if anyone left alive could still care about aesthetics.",
```

TheShadowFerret commented on data/json/items/armor/legs_clothes.json:
```diff
@@ -606,7 +606,377 @@
+      {
+        "id": "pants_blue",
+        "name": { "str": "navy-blue pants", "str_pl": "pairs of navy-blue pants" },
+        "description": "A comfortable set of navy blue pants with large pockets: these trousers    are slightly thicker and warmer than regular jeans.",
```
```suggestion
        "description": "A comfortable set of navy blue pants with large pockets, slightly thicker and warmer than regular jeans.",
```

TheShadowFerret commented on data/json/items/armor/legs_clothes.json:
```diff
@@ -606,7 +606,377 @@
+      {
+        "id": "pants_grey",
+        "name": { "str": "grey pants", "str_pl": "pairs of grey pants" },
+        "description": "a set of storm gray pants speckled with dark stains so deeply assimilated into the fabric that they’re nigh-irremovable. You vehemently hope that they’re just the mortal remains of some spilled Beveridge.",
```
```suggestion
        "description": "A set of storm gray pants speckled with dark stains so deeply assimilated into the fabric that they’re nigh-irremovable.  You vehemently hope that they’re just the mortal remains of some spilled beverage.",
```
~~Who's Beveridge~~

TheShadowFerret commented on data/json/items/armor/legs_clothes.json:
```diff
@@ -606,7 +606,377 @@
+      {
+        "id": "pants_od",
+        "name": { "str": "olive-drab pants", "str_pl": "pairs of olive-drab pants" },
+        "description": "While these aren’t US-issued military trousers or even civilian tactical pants, this article of clothing possesses a blend of dark greens, browns, and blacks identical to the camouflage pattern of Vietnam-era army uniforms.  Slightly warmer than regular jeans, it’s a decent piece of foe-military kit for those unwilling to shill out on a pair of actual BDU pants.",
```
```suggestion
        "description": "While these aren’t US-issued military trousers or even civilian tactical pants, this article of clothing possesses a blend of dark greens, browns, and blacks identical to the camouflage pattern of Vietnam-era army uniforms.  Slightly warmer than regular jeans, it’s a decent piece of faux-military kit for those unwilling to shill out on a pair of actual BDU pants.",
```

TheShadowFerret commented on data/json/items/armor/legs_clothes.json:
```diff
@@ -606,7 +606,377 @@
+      {
+        "id": "pants_arctic",
+        "name": { "str": "arctic-camo pants", "str_pl": "pairs of arctic-camo pants" },
+        "description": "Featuring a camouflage pattern consisting of white, shades of gray, and a slight tinge of brown, this pair of rugged pants is fit for snowy tundra and poler regions—or it would be, had the garment not been only moderately warmer than jeans.  You have yet to grasp their point for existing.",
```
```suggestion
        "description": "Featuring a camouflage pattern consisting of white, shades of gray, and a slight tinge of brown, this pair of rugged pants is fit for snowy tundra and polar regions—or it would be, had the garment not been only moderately warmer than jeans.  You have yet to grasp their point for existing.",
```

TheShadowFerret commented on data/json/items/armor/legs_clothes.json:
```diff
@@ -606,7 +606,377 @@
+      {
+        "id": "pants_plum",
+        "name": { "str": "plum-purple trackpants", "str_pl": "pairs of plum-purple trackpants" },
+        "description": "With its fabric dyed a relatively pretty shade of deep reddish purple, this pair of pants not only looks good but also provides a level of practicality only befitting its role as sports wear, coming equipped with an elastic waistband and zip-up pockets and constructed from a stretchy material that clings to the hips and legs.",
```
>a stretchy material that clings to the hips and legs

That doesn't sound like cotton to me 👀

TheShadowFerret commented on data/json/items/armor/legs_clothes.json:
```diff
@@ -606,7 +606,377 @@
+      {
+        "id": "pants_fuchsia",
+        "name": { "str": "fuchsia pants", "str_pl": "pairs of fuchsia pants" },
+        "description": "a bright, fashionable, and charming-looking pair of reddish purple trousers, with two deep pockets and a construction that favors heat retention more than regular jeans.",
```
```suggestion
        "description": "A bright, fashionable, and charming-looking pair of reddish purple trousers, with two deep pockets and a construction that favors heat retention more than regular jeans.",
```

DoctorBoomstick committed:
Update data/json/items/armor/boots.json

Co-authored-by: TheShadowFerret <99621099+TheShadowFerret@users.noreply.github.com>

DoctorBoomstick committed:
Update data/json/items/armor/coats.json

Co-authored-by: TheShadowFerret <99621099+TheShadowFerret@users.noreply.github.com>

DoctorBoomstick committed:
Update data/json/items/armor/coats.json

Co-authored-by: TheShadowFerret <99621099+TheShadowFerret@users.noreply.github.com>

DoctorBoomstick committed:
Update data/json/items/armor/coats.json

Co-authored-by: TheShadowFerret <99621099+TheShadowFerret@users.noreply.github.com>

DoctorBoomstick committed:
Update data/json/items/armor/coats.json

Co-authored-by: TheShadowFerret <99621099+TheShadowFerret@users.noreply.github.com>

DoctorBoomstick committed:
Update data/json/items/armor/legs_clothes.json

Co-authored-by: TheShadowFerret <99621099+TheShadowFerret@users.noreply.github.com>

DoctorBoomstick committed:
Update data/json/items/armor/legs_clothes.json

Co-authored-by: TheShadowFerret <99621099+TheShadowFerret@users.noreply.github.com>

DoctorBoomstick committed:
Update data/json/items/armor/legs_clothes.json

Co-authored-by: TheShadowFerret <99621099+TheShadowFerret@users.noreply.github.com>

DoctorBoomstick committed:
Update data/json/items/armor/legs_clothes.json

Co-authored-by: TheShadowFerret <99621099+TheShadowFerret@users.noreply.github.com>

DoctorBoomstick committed:
Update data/json/items/armor/coats.json

Co-authored-by: TheShadowFerret <99621099+TheShadowFerret@users.noreply.github.com>

DoctorBoomstick committed:
Update data/json/items/armor/coats.json

Co-authored-by: TheShadowFerret <99621099+TheShadowFerret@users.noreply.github.com>

DoctorBoomstick committed:
Update data/json/items/armor/legs_clothes.json

Co-authored-by: TheShadowFerret <99621099+TheShadowFerret@users.noreply.github.com>

DoctorBoomstick committed:
Makes pants that’s made from cotton actually sound like it’s made from cotton rather than defying the laws of material properties.

Termineitor244 commented:
Haha, I love it! I specially like that you included flag variants for some of these items, you should replace the sneakers of the patriot profession (when female) with your new flag heels, as well as changing the flag pants with your new variant.

On that note, you should probably obsolete the flag pants item I added some time ago and migrate them with your new flag pants variant (And if you do, please tell the people in the tileset discord to remember to move the sprite for the pants to the new variant), when I added them, variants were still a very recent thing so I did not think of making them a variant until after my costume PRs were merged haha.

Oh, and if you want, you can change the color of the item symbol itself, for when a specific colored variant is spawned. Check the aprons variants for examples.

Termineitor244 commented:
Haha, I love it! I specially like that you included flag variants for some of these items, you should replace the sneakers of the patriot profession (when female) with your new flag heels, as well as changing the flag pants with your new variant.

On that note, you should probably obsolete the flag pants item I added some time ago and migrate them with your new flag pants variant (And if you do, please tell the people in the tileset discord to remember to move the sprite for the pants to the new variant), when I added them, variants were still a very recent thing so I did not think of making them a variant until after my costume PRs were merged haha.

Oh, and if you want, you can change the color of the item symbol itself, for when a specific colored variant is spawned. Check the aprons variants for examples.

bombasticSlacks commented:
approved but gonna let you mull over the comments before merging

DoctorBoomstick commented:
> Haha, I love it! I specially like that you included flag variants for some of these items, you should replace the sneakers of the patriot profession (when female) with your new flag heels, as well as changing the flag pants with your new variant.
> 

I completely forgot that flag pants were already a thing. Oops. I’ll swap them around for the glorious word of liberbooty, and do the same for the sneakers. This makes me think that I should probably also take the time to make freedom boots while I’m at it.
> On that note, you should probably obsolete the flag pants item I added some time ago and migrate them with your new flag pants variant (And if you do, please tell the people in the tileset discord to remember to move the sprite for the pants to the new variant), when I added them, variants were still a very recent thing so I did not think of making them a variant until after my costume PRs were merged haha.
> 


Roger that, will do.
> Oh, and if you want, you can change the color of the item symbol itself, for when a specific colored variant is spawned. Check the aprons variants for examples.


Adding this to the list of stuff that needs adding.

DoctorBoomstick committed:
last tweeks, adds colours to all the pants and removes the flag pants.

DoctorBoomstick committed:
fixes bug.

DoctorBoomstick committed:
retrigger checks

Maleclypse commented on data/mods/Xedra_Evolved/itemgroups/itemgroups.json:
```diff
@@ -186,7 +186,6 @@
       [ "sleeveless_trenchcoat_fur", 10 ],
       [ "sleeveless_duster_fur", 10 ],
       [ "sleeveless_trenchcoat_faux_fur", 10 ],
-      [ "coat_fur_sf", 50 ],
```
What is this and why is it getting removed in a pants pr?

DoctorBoomstick replied:
> adds cosmetic variants for various items, including regular pants, fur coats,

Apart for pants, which is the main element of this PR, I also bundled in variants for fur coats. “coat_fur_sf” was the id for the sable fur coat, which has been iced and made into a variant of the basic fur coat item.
